src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c

```diff
@@ -783,3 +795,52 @@ void reset_notify_client_ctx(struct xocl_dev *xdev)
 	atomic_set(&xdev->needs_reset,0);
 	wmb();
 }
+
+int xocl_p2p_enable_ioctl(struct drm_device *dev,
+			  void *data,
+			  struct drm_file *filp)
+{
+	struct xocl_dev *xdev = dev->dev_private;
+	struct pci_dev *pdev = xdev->core.pdev;
+	int ret, p2p_bar, enable;
+	u32 size;
+
+	enable = ((struct drm_xocl_p2p_enable *)data)->enable;
+
+	p2p_bar = xocl_get_p2p_bar(xdev);
+	if (p2p_bar < 0) {
+		xocl_err(&pdev->dev, "p2p bar is not configurable");
+		ret = -EINVAL;
+		goto failed;
+	}
+
```

Comment:
I think you should be able to skip below operation if the current state is the same as what user is asking for...

Reply:
return EACCES if error